codegen-core/src/test/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/InlineDependencyTest.kt

```diff
@@ -28,17 +33,66 @@ internal class InlineDependencyTest {
     @Test
     fun `locate dependencies from the inlineable module`() {
         val dep = InlineDependency.idempotencyToken()
-        val testWriter = RustWriter.root()
-        testWriter.addDependency(CargoDependency.FastRand)
-        testWriter.withModule(dep.module.copy(rustMetadata = RustMetadata(visibility = Visibility.PUBLIC))) {
-            dep.renderer(this)
+        val testWriter = TestWorkspace.testProject()
+        testWriter.unitTest {
+            rustTemplate(
+                """
+                use #{idempotency}::uuid_v4;
+                let res = uuid_v4(0);
+                assert_eq!(res, "00000000-0000-4000-8000-000000000000");
+
+                """,
+                "idempotency" to dep.asType(),
+            )
         }
-        testWriter.compileAndTest(
-            """
-            use crate::idempotency_token::uuid_v4;
-            let res = uuid_v4(0);
-            assert_eq!(res, "00000000-0000-4000-8000-000000000000");
-            """,
-        )
+        testWriter.compileAndTest()
```

Comment:
I thought we were rewriting tests to use the more modern `val project = TestWorkspace.testProject(symbolProvider)` machinery if we had to modify an old test file.

Reply:
(this does update to use test project, but fails to rename, fixed)